apps/meteor/app/utils/lib/restrictions.ts

```diff
@@ -17,37 +15,45 @@ const fileUploadMediaBlackList = function (customBlackList: string): string[] |
 		return;
 	}
 
-	return _.map(blacklist.split(','), (item) => item.trim());
+	return blacklist.split(',').map((item) => item.trim());
 };
 
 const isTypeOnList = function (type: string, list: string[]): boolean | undefined {
-	if (_.contains(list, type)) {
+	if (list.includes(type)) {
 		return true;
 	}
 	const wildCardGlob = '/*';
-	const wildcards = _.filter(list, (item) => {
+	const wildcards = list.filter((item) => {
 		return item.indexOf(wildCardGlob) > 0;
 	});
-	if (_.contains(wildcards, type.replace(/(\/.*)$/, wildCardGlob))) {
+	if (wildcards.includes(type.replace(/(\/.*)$/, wildCardGlob))) {
 		return true;
 	}
 };
 
-export const fileUploadIsValidContentTypeFromSettings = function (type: string, customWhiteList: string, customBlackList: string): boolean {
+export const fileUploadIsValidContentTypeFromSettings = function (
+	type: string | undefined,
+	customWhiteList: string,
+	customBlackList: string,
+): boolean {
 	const blackList = fileUploadMediaBlackList(customBlackList);
 	const whiteList = fileUploadMediaWhiteList(customWhiteList);
 
-	if (!type && blackList) {
+	if (blackList && type && isTypeOnList(type, blackList)) {
 		return false;
 	}
 
-	if (blackList && isTypeOnList(type, blackList)) {
+	if (whiteList && type && isTypeOnList(type, whiteList)) {
+		return true;
+	}
+
+	if (!type && whiteList) {
 		return false;
 	}
```

Comment:
```suggestion
	if (whiteList) {
		return isTypeOnList(type, whiteList);
	}
```
Is believe this works just the same, doesn't it? Looks easier to read

Reply:
It does, but I had to make some smalls adjustments to `isTypeOnList` func ... Thanks for your feedback